tests/scripts/test_build_macos.py

```
@@ -0,0 +1,54 @@
+    )
+
+
+def test_build_nuitka_command_raises_when_manifest_missing(tmp_path: Path) -> None:
+    """Missing manifest.json must fail the build instead of omitting the flag."""
+    build_macos = _load_build_macos_module()
+    main_entry = tmp_path / "lintro" / "__main__.py"
+    main_entry.parent.mkdir(parents=True)
+    main_entry.write_text("", encoding="utf-8")
+
+    with patch.object(build_macos, "PROJECT_ROOT", tmp_path):
+        with pytest.raises(FileNotFoundError, match="manifest.json"):
```

Ruff (0.15.13)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)